File: administratorapp/lib/domain/entities/Movimiento.dart
import 'package:administratorapp/domain/entities/Producto.dart';
import 'package:administratorapp/domain/entities/Usuario.dart';

enum TipoMovimiento {
  INGRESO,
  EGRESO,
}

class Movimiento {
  final int movimientoid;
  final TipoMovimiento tipoMovimiento;
  final int cantidad;
  final DateTime fecha;
  final Producto producto;
  final Usuario usuario;

  Movimiento({
    required this.movimientoid,
    required this.tipoMovimiento,
    required this.cantidad,
    required this.fecha,
    required this.producto,
    required this.usuario,
  });
}

File: api/Models/Comercial/Movimiento.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using api.Models.Cuenta;

namespace api.Models.Comercial;

public class Movimiento
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Movimientoid { get; set; }

    public TipoMovimiento Tipomovimiento { get; set; }

    public int Cantidad { get; set; }

    public DateTime Fecha { get; set; }

    public int Idproducto { get; set; }

    public int Idusuario { get; set; }
    [ForeignKey("Idproducto")] public virtual Productos Producto { get; set; } = null!;
    [ForeignKey("Idusuario")] public virtual Usuario Usuario { get; set; } = null!;
}